smart_open/tests/test_smart_open.py

```diff
@@ -200,92 +201,173 @@ def test_http_pass(self):
+    def test_http_xz_query(self):
+        """Can open xz via http with a query appended to URI?"""
+        self._test_compressed_http(".xz", True)
 
-        m = hashlib.md5(smart_open_object.read())
-        # decompress the gzip and get the same md5 hash
-        self.assertEqual(m.hexdigest(), expected_hash)
 
-    @responses.activate
-    def test_http_bz2(self):
-        """Can open bz2 via http?"""
-        test_string = b'Hello World Compressed.'
-        #
-        # TODO: why are these tests writing to temporary files?  We can do the
-        # bz2 compression in memory.
-        #
-        with tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False) as infile:
-            test_file = infile.name
+class SmartOpenFileObjTest(unittest.TestCase):
+    """
+    Test passing raw file objects.
+    """
 
-        with smart_open.smart_open(test_file, 'wb') as outfile:
-            outfile.write(test_string)
+    class IOReadWrapper(object):
+        def __init__(self):
+            self.fobj = six.BytesIO()
 
-        with open(test_file, 'rb') as infile:
-            compressed_data = infile.read()
+        def read(self, size=-1):
+            return self.fobj.read(size)
 
-        if os.path.isfile(test_file):
-            os.unlink(test_file)
+        def __enter__(self):
+            return self
 
-        responses.add(responses.GET, "http://127.0.0.1/data.bz2", body=compressed_data, stream=True)
-        smart_open_object = smart_open.smart_open("http://127.0.0.1/data.bz2")
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            self.close()
 
-        # decompress the bzip2 and get the same md5 hash
-        self.assertEqual(smart_open_object.read(), test_string)
+        def close(self):
+            self.fobj.close()
 
-    @responses.activate
-    def test_http_xz(self):
-        """Can open xz via http?"""
-        test_string = b'Hello World Compressed.'
-        #
-        # TODO: why are these tests writing to temporary files?  We can do the
-        # lzma compression in memory.
-        #
-        with tempfile.NamedTemporaryFile('wb', suffix='.xz', delete=False) as infile:
-            test_file = infile.name
-
-        with smart_open.smart_open(test_file, 'wb') as outfile:
-            outfile.write(test_string)
+    class IOWriteWrapper(object):
+        def __init__(self):
+            self.fobj = six.BytesIO()
 
-        with open(test_file, 'rb') as infile:
-            compressed_data = infile.read()
+        def write(self, b):
+            return self.fobj.write(b)
 
-        if os.path.isfile(test_file):
-            os.unlink(test_file)
+        def close(self):
+            self.fobj.close()
 
-        responses.add(responses.GET, "http://127.0.0.1/data.xz", body=compressed_data, stream=True)
-        smart_open_object = smart_open.smart_open("http://127.0.0.1/data.xz")
+    def test_read(self):
+        file = self.IOReadWrapper()
```

Comment:
@mpenkov FWIW `file` is *not* a keyword. Proof:

```python
>>> import keyword
>>> keyword.iskeyword("file")
```

It is a *built-in type* and only in Python2:

```python
>>> import builtins
>>> "file" in dir(builtins)
```

But yeah, I am just being pedantic - ofc I will change the name :smile: It's been a long time since I last worked with py2.